File: atcoder.jp/contests/abc155/tasks/abc155_b/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

const even = (n: number) => n % 2 === 0;

const main = () => {
  const N = +input();
  const A = input().split(" ").map(Number);

  const ans = A.filter(even).every((a) => a % 3 === 0 || a % 5 === 0);

  console.log(ans ? "APPROVED" : "DENIED");
};

main();

File: atcoder.jp/contests/abc155/tasks/abc155_b/main.hs
main :: IO ()
main = do
  getLine
  as <- map read . words <$> getLine

  let ans = and [a `mod` 3 == 0 || a `mod` 5 == 0 | a <- as, even a]

  putStrLn $ if ans then "APPROVED" else "DENIED"
